File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False] * n
    result = 0
    
    def DFS(com):
        visited[com] = True
        
        for i in range(n):
            if not visited[i] and computers[com][i] == 1:
                DFS(i)
    
    for i in range(n):
        if not visited[i]:
            DFS(i)
            result += 1
    return result
        

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let visited = new Array(n).fill(false);
    let result = 0;
    
    const DFS = (com) => {
        visited[com] = true;
        
        for(let i = 0 ; i < n ; i ++){
            if(computers[com][i] === 1 && !visited[i]) DFS(i);
        }
    }
     
    for(let i = 0 ; i < n ; i++){
        if(!visited[i]){
            DFS(i);
            result++;
        }
    }
    return result;
}